functions_and_scope_lab.js

```diff
@@ -0,0 +1,286 @@
+//
+// | Number Grade | Equivalent Letter Grade |
+// | :--------: | :---------: |
+// | 100 | A+ |
+// | 90 - 99 | A |
+// | 80 - 89 | B |
+// | 70 - 79 | C |
+// | 65 - 69 | D |
+// | Below 65 | F |
+//
+//
+// Example:
+//
+// Input: `88`
+//
+// Output: `"B"`
+
+const getLetterGrade = (numGrade) =>{
+  if(numGrade >= 90 && numGrade <100){
+    return 'A';
+  } else if(numGrade<90 && numGrade>= 80){
+    return 'B';
+  } else if (numGrade < 80 && numGrade >= 70) {
+    return 'C';
+  }else if(numGrade < 70 && numGrade >= 65){
+    return 'D';
+  }else{
+    return 'F';
+  }
+}
+let finalGrade = getLetterGrade(88);
+console.log(finalGrade)
+
+
+// ## Question 6
+//
+// Implement a function named `repeatLog` that takes a string and a number as parameters. The function should log `message` `count` number of times.
+//
+// Example:
+//
+// Input: `"+", 10`
+//
+// Output: `++++++++++`
+
+//
+const repeatLog = (str,inc) =>{
+  let output = '';
+  for(let i = 0; i < inc; i++){
+      output += str
+  }
+  console.log(output)
+}
+repeatLog('+',10);
+
+// ## Question 7
+
+// Write a function named `first` that takes a number and returns an array with all the numbers from 1 to n
+//
+//
+// Example:
+//
+// Input: `first(3)`
+//
+// Output: `[1, 2, 3]`
+//
+
+const firstThree = (n) =>{ // n is the number to create the array from
+let emptyArr = [] //create empty array to push all the values from 0 to n into the arrays
+for(let i = 1; i <= n; i++){ //the length is the value of the number
+  emptyArr.push(i) //.push is the plus equal for the array, it essentially adds elements into an array, do it in i because of the loop through the number.
+}
+return emptyArr // this is returned because it has all of the variables in the emptyArr
+}
+let something = firstThree(5)
+console.log(something)
+
+// ## Question 8
+//
+// Write a function that logs the numbers from 1 to x, except:
+//
+// If the number if a multiple of 3, log `"Fizz"` instead of the number
+// If the number is a multiple of 5, log `"Buzz"` instead of the number
+// If the number is a multiple of 3 AND 5, log `"FizzBuzz"` instead of the number
+// Your function should take in one parameter: x (the number to count up to)
+
+const logFizzBuzz = (x) =>{
+  for(let y = 1; y <= 100; y++){
+    if(x % 3 === 0 && x % 5 === 0){
+    return 'FizzBuzz';
+    } else if ( x % 5 === 0){
+      return 'Buzz';
+    } else if ( x % 3 === 0 ){
+      return 'Fizz';
+    }else{
+      return x;
+    }
+}
+}
+let finalFB = logFizzBuzz(15)
+console.log(finalFB)
+
+// ## Question 9
+//
+// Write a function named `reverse` that takes an array as a parameter. The function should return an array in reverse order.
+//
+//
+// Example:
+//
+// Input: `reverse([1, 2, 3])`
+//
+// Output: `[3, 2, 1]`
+//
+const reverseArray = () =>{
+
+
+}
+
+// ## Question 10
+//
+// Write a function that prints out the most frequently occurring element in an array.
```

Comment:
Try to solve this problem when you have chance. This is one of whiteboarding questions that some people might ask about.